db-tests/src/test/java/gov/nasa/jpl/aerie/database/ExternalEventTests.java

```diff
@@ -0,0 +1,1922 @@
+package gov.nasa.jpl.aerie.database;
+
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestInstance;
+
+import java.io.IOException;
+import java.sql.Connection;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.ArrayList;
+
+import org.postgresql.util.PSQLException;
+
+import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertThrowsExactly;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+@SuppressWarnings("SqlSourceToSinkFlow")
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+public class ExternalEventTests {
+
+  //region Generic Database Testing Setup
+  private DatabaseTestHelper helper;
+  private MerlinDatabaseTestHelper merlinHelper;
+  private Connection connection;
+
+  @AfterEach
+  void afterEach() throws SQLException {
+    helper.clearSchema("merlin");
+  }
+
+  @BeforeAll
+  void beforeAll() throws SQLException, IOException, InterruptedException {
+    helper = new DatabaseTestHelper("external_event", "Activity Directive Changelog Tests");
+    connection = helper.connection();
+    merlinHelper = new MerlinDatabaseTestHelper(connection);
+  }
+
+  @AfterAll
+  void afterAll() throws SQLException, IOException, InterruptedException {
+    helper.clearSchema("merlin");
+    helper.close();
+    connection = null;
+    helper = null;
+  }
+  //endregion
+
+
+  //region Commonly Repeated Variables
+  // Source Type (st)
+  final String st = "Test";
+
+  // Derivation Group (dg)
+  final String dg = "Test Default";
+
+  // Event Type (et)
+  final String et = "Test";
+
+  // Metadata/Properties (mt)
+  final String mt = "{}";
+
+  // Created At (ca)
+  final String ca = "2024-01-01T00:00:00Z";
+  //endregion
+
+
+  //region Records
+  protected record ExternalEvent(String key, String event_type_name, String source_key, String derivation_group_name, String start_time, String duration, String properties) {}
+  protected record ExternalSource(String key, String source_type_name, String derivation_group_name, String valid_at, String start_time, String end_time, String created_at, String metadata){}
+  //endregion
+
+
+  //region Helper Functions
+  /**
+   * A simple function to compare a SQL result (res) for a given key (key) against a list of expected entries (expected)
+   */
+  void compareLists(String[] expected, ResultSet res, String key) throws SQLException {
+    ArrayList<String> actual = new ArrayList<>();
+    while (res.next()) {
+      actual.add(res.getString(key));
+    }
+
+    assertEquals(expected.length, actual.size());
+    for (int i = 0; i < expected.length; i++) {
+      assertEquals(expected[i], actual.get(i));
```

Comment:
Is comparing the order important? If not, then replace this for loop with `assertTrue(actual.containsAll(expected))`